configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -137,7 +138,7 @@ public ConfigurationFactory(Map<String, String> properties, WarningsMonitor warn
     {
         this.properties = ImmutableMap.copyOf(properties);
         this.warningsMonitor = warningsMonitor;
-        this.monitor = monitor;
+        ConfigurationFactory.monitor = monitor;
```

Comment:
We shouldn't set a static field from a constructor. It's confusing and not thread safe.

Instead, we should refactor the tests so that `Problems.Monitor` is not needed. It appears that the tests could access `Problems` directly, but you'll need to try refactoring it and see what it looks like.

Reply:
Changed it so that it's not static anymore.
I did try to remove monitor but it require more code changes that I wanted to fit in this PR